common/buildcraft/builders/tile/TileArchitect_Neptune.java

```diff
@@ -113,26 +111,33 @@ public void onPlacedBy(EntityLivingBase placer, ItemStack stack) {
 
     @Override
     public void update() {
+        deltaProgress.tick();
+
         if (worldObj.isRemote) {
             return;
         }
         if (scanning && isValid) {
-            scanMultipleBlocks();
+            int target = shouldScanDetails ? 30 : 10;
```

Comment:
If the architect takes any more than a single tick for a block then its going to take a huge amount of time to scan the entire blueprint, which is something I'd like to avoid.